stanford-car-classification/README.md

```diff
@@ -0,0 +1,288 @@
+
+The Hyperparameter Optimization (HPO) conducted is a layer on top of the model tuning as explicated above.
+
+### Pre-requistes
+
+The HPO uses [SigOpt](https://sigopt.com/)'s [Multitask](https://app.sigopt.com/docs/overview/multitask) feature as the optimizer as well as [Orchestrate](https://app.sigopt.com/docs/orchestrate) to manage AWS clusters. 
+In order to be able to run the optimization, please set up your SigOpt account and walk through the [Orchestrate tutorial](https://app.sigopt.com/docs/orchestrate/tutorial/1).
+At the end of the tutorial, you should have an Orchestrate specific virtualenvironment which we will use later.
+
+### CommandLine Interface
+
+```
+python run_resnet_training_cli.py
+--path_images <path to parent directory of car_ims folder>
+--path_data <path to cars_meta.mat> 
+--path_labels <path to cars_annos.meta>
+[--path_model_checkpoint <path to model checkpointing directory, default: No check-pointing>] 
+[--checkpoint_frequency <frequency to generate PyTorch checkpoint files>, default: No check-pointing]
+--model {ResNet18 | ResNet50}
+--epochs <number of epochs to train model>
+--validation_frequency <frequency to run validation during training> 
+--number_of_classes <number of labels in data set>
+--data_subset <subset of data to be used [0, 1.0]>
+ {--freeze_weights | --no-freeze_weights}
+
+```
+
+To run HPO on tuning the fully connected layer: --freeze_weights must be included
+
+To run HPO on fine tuning the whole network: --no-freeze_weights must be included
+
+The data_subset option is used to specify the fraction of the Stanford Cars dataset to use (data_subset = 0.5 means 50% of the data is used).
+A 20% validation split is applied to the data used.
+
+Logs will be outputted to the working directory. We suggest to not checkpoint during hyperparameter optimization.
+
+SigOpt suggests values for the following hyperparameters:
+* batch size
+* learning rate
+* learning rate scheduler
+* scheduler rate
+* weight decay
+* momentum
+* nesterov
+
+The bounds of these hyperparameters are specified in the Orchestrate experiment configuration file `orchestrate_stanford_cars_tuning_config.yml`.
+
+Example:
+
+```buildoutcfg
+source ./stanford-car-classification-venv/bin/activate
+python run_resnet_training_cli.py --path_images ./stanford-car-classification/data/ --path_data ./stanford-car-classification/data/cars_annos.mat --path_labels ./stanford-car-classification/data/cars_meta.mat --model ResNet18 --epochs 2 --validation_frequency 10 --data_subset 1.0  --number_of_classes 196 --no-freeze_weights
+```
+
+### Cluster Configuration
+
+As seen in the [Orchestrate tutorial](https://app.sigopt.com/docs/orchestrate/tutorial/1), a cluster configuration file is necessary to deploy a cluster.
+The following snippet is an example cluster configuration `orchestrate_cluster_deploy_sample.yml` that deploys 2 p2.xlarge EC2 instances on AWS.
+
+```buildoutcfg
+
+# must be a .yml file
```

Comment:
I find it helpful to annotate something like  `# orchestrate_cluster_deploy_sample.yml` to illustrate the filename. Though, why not include the cluster configuration file as a file in the repository, rather than written out in the readme?

Reply:
Oh I see, it has been included, but it looks like it's not referred to by the README?